pkg/build/metrics/prometheus/metrics.go

```diff
@@ -77,72 +77,79 @@ func (bc *buildCollector) Collect(ch chan<- prometheus.Metric) {
 		return
 	}
 
-	// since we do not collect terminal build metrics on a per build basis, collectBuild will return counts
-	// to be added to the total amount posted to prometheus
-	var cancelledCount, completeCount, errorCount, newCount, pendingCount, runningCount int
-	reasons := map[string]int{}
+	// collectBuild will return counts for the build's phase/reason tuple,
+	// and counts for these tuples be added to the total amount posted to prometheus
+	counts := map[string]map[string]int{
+		cancelledPhase: map[string]int{},
+		completePhase:  map[string]int{},
+		errorPhase:     map[string]int{},
+		failedPhase:    map[string]int{},
+		newPhase:       map[string]int{},
+		pendingPhase:   map[string]int{},
+		runningPhase:   map[string]int{},
+	}
 	for _, b := range result {
-		cc, cp, ec, nc, pc, rc, r := bc.collectBuild(ch, b)
-		for key, value := range r {
-			reasons[key] = reasons[key] + value
+		r := bc.collectBuild(ch, b)
+		for phase, reasonMap := range r {
+			currentPhaseMap := counts[phase]
+			for reason, count := range reasonMap {
+				if _, ok := currentPhaseMap[reason]; !ok {
+					currentPhaseMap[reason] = 0
+				}
+				currentPhaseMap[reason] = currentPhaseMap[reason] + count
+			}
 		}
-		cancelledCount = cancelledCount + cc
-		completeCount = completeCount + cp
-		errorCount = errorCount + ec
-		newCount = newCount + nc
-		pendingCount = pendingCount + pc
-		runningCount = runningCount + rc
-	}
-	// explicitly note there are no failed builds
-	if len(reasons) == 0 {
-		addCountGauge(ch, buildCountDesc, failedPhase, "", float64(0))
 	}
-	for reason, count := range reasons {
-		addCountGauge(ch, buildCountDesc, failedPhase, reason, float64(count))
+
+	for phase, reasonMap := range counts {
+		// explicitly note there are no items of this phase
+		// we do not worry about posting 0 for a particular reason of a phase
+		if len(reasonMap) == 0 {
```

Comment:
why is this necessary, out of interest?

Reply:
necessary, no

but in experimenting with the console, I found it more intuitive when I could explicitly see that there were say 0 failed builds (vs. there simply being an absence of metrics with the label failed)

with the build polling, if there are no failed builds, etc, no metrics get registered

Reply:
that said ... i'm fine with removing, and having consumers use `count(...) == 0` to get the equivalent